File: examples/numexpr-edsl/core/src/Data/NumExpr.erl
-module(data_numExpr@foreign).
-export([numPow/2, numSin/1, numCos/1, numExp/1, numLog/1, numSqrt/1]).

numPow(X, Y) -> math:pow(X, Y).
numSin(X) -> math:sin(X).
numCos(X) -> math:cos(X).
numExp(X) -> math:exp(X).
numLog(X) -> math:log(X).
numSqrt(X) -> math:sqrt(X).

File: examples/numexpr-edsl/core/src/Data/NumExpr.js
// JS backend shim for Data.NumExpr's transcendental primitives (node/ workspace).
// The JS package set keeps these on Math.*; PS curried Number->Number->Number
// for numPow.
export const numPow = a => b => Math.pow(a, b);
export const numSin = Math.sin;
export const numCos = Math.cos;
export const numExp = Math.exp;
export const numLog = Math.log;
export const numSqrt = Math.sqrt;
